src/Servers/Kestrel/Core/src/Internal/Http2/FlowControl/StreamOutputFlowControl.cs

```diff
@@ -24,6 +24,11 @@ public StreamOutputFlowControl(OutputFlowControl connectionLevelFlowControl, uin
 
         public bool IsAborted => _connectionLevelFlowControl.IsAborted || _streamLevelFlowControl.IsAborted;
 
+        public void Reset()
+        {
+            _streamLevelFlowControl.Reset();
```

Comment:
_currentConnectionLevelAwaitable needs to be completed and then nulled out. This should have happened already, but we should at least have an assert. Are we really gaining much by pooling the flow control classes?

Reply:
We save allocating them per-request.

Reply:
I know, but there are other things that get allocated per request like strings. I'm just worried about introducing bugs with this pooling, whether it's not fully resetting things correctly, or resetting objects while they're still in use.